clang_build/compiler.py

```diff
@@ -135,26 +134,28 @@ def _dialect_exists(self, year, clangpp):
         return True
 
     @_lru_cache(maxsize=1)
-    def _get_max_supported_compiler_dialect(self, clangpp):
+    def _get_max_supported_compiler_dialect(self):
         """Check the maximally supported C++ dialect.
 
-        Parameters
-        ----------
-        clangpp : :any:`pathlib.Path`
-            Path to the clang++ executable
-
         Returns
         -------
         str
             Flag string of the latest supported dialect
 
         """
-        supported_dialects = []
-        for dialect in range(30):
-            if self._dialect_exists(dialect, clangpp):
-                supported_dialects.append(dialect)
-
-        if supported_dialects:
-            return self._get_dialect_flag(max(supported_dialects))
-        else:
-            return self._get_dialect_flag(98)
+        try:
```

Comment:
why the `try`/`except` here?

Reply:
Well clang build fails with the command given, it just so happens that it also spits out the valid values for "-std". This part is slightly streamlines with the next compiler pull request.

Reply:
Ah now I see what you're doing - interesting solution